scipy/stats/_axis_nan_policy.py

```diff
@@ -391,17 +443,27 @@ def hypotest_fun_out(*samples, **kwds):
             vectorized = True if 'axis' in params else False
             axis = kwds.pop('axis', default_axis)
             nan_policy = kwds.pop('nan_policy', 'propagate')
+            keepdims = kwds.pop("keepdims", False)
             del args  # avoid the possibility of passing both `args` and `kwds`
 
             # convert masked arrays to regular arrays with sentinel values
             samples, sentinel = _masked_arrays_2_sentinel_arrays(samples)
 
             # standardize to always work along last axis
             if axis is None:
+                # when axis=None, take the maximum of all dimensions since
+                # all the dimensions are reduced.
+                if samples:
+                    max_dims = max([sample.ndim for sample in samples])
+                    reduced_axes = np.arange(max_dims)
```

Comment:
Hmm we write that the output array will be broadcastable with the inputs, but if we use `max` here, that's not true. Maybe we want `min`, actually?

Reply:
Good catch, this got me thinking! I think both should work: since we are anyways going to return an array of shape `(1,1,1,...)`, if all other arrays broadcast with each other, they must also broadcast with an array of all 1s. Even if we return array with `min` ndims, it would be filled with 1s on the right when broadcasting. Does this sound right or did you have any case in mind where `min` is desirable over `max`?

Reply:
I see what you're saying.
You are right that all of the input and output arrays will broadcast together either way.
I guess my thought was that if you use `min` instead of `max`, then the _output_ array will broadcast _to_ the shape of any of the input arrays individually.
On the other hand, if you use `max`, then even though the input and output arrays are "broadcastable", the output array will not necessarily broadcast to the shape of each of the input arrays.
Does that make sense?
Here's an example:
```python3
x = np.random.rand(3, 4)
y = np.random.rand(5, 3, 1)

out1 = (x*y).sum(axis=None).reshape((1, 1))  # `min`
np.broadcast(x, y, out1)   # no problem
np.broadcast_to(out1, x.shape)   # no problem
np.broadcast_to(out1, y.shape)   # no problem

out1 = (x*y).sum(axis=None).reshape((1, 1, 1))  # `max`
np.broadcast(x, y, out1)   # no problem
np.broadcast_to(out1, x.shape)   # problem!
np.broadcast_to(out1, y.shape)   # no problem
```
That said, I think either way satisfies the statement:
> With this option, the result will broadcast correctly against the input array.

because "against" doesn't imply a direction very strongly. 

My thought is just that `min` is slightly more flexible than `max`.

On the other hand, the argument name `keepdims` suggests that we keep all the dimensions we have before the statistic is taken. 
```out1 = (x*y).sum(axis=None, keepdims=True).shape  # `(1, 1, 1)`
```
So that's an argument in favor of `max`.

Can you find any two-"sample" NumPy functions we can use as precedent? (This only matters when there are multiple "samples" that might be different shapes.) I tried for a bit without luck.

Reply:
> Does that make sense?

Makes sense but I have one more argument in favor of `max` :)

> On the other hand, the argument name `keepdims` suggests that we keep all the dimensions we have before the statistic is taken.

To me, `max` makes intuitive sense because of this argument. Also, with this, the behavior of `axis=None` and `axis=(0, 1, 2, ...)` matches; both fill in max number of 1s. For example:

```py
>>> import numpy as np
>>> from scipy import stats
>>> x = np.random.rand(3, 4, 5)
>>> y = np.random.rand(4, 5)
>>> res = stats.kruskal(x, y, axis=None, keepdims=True)
>>> res[0].shape
(1, 1, 1)
>>> res = stats.kruskal(x, y, axis=(0, 1, 2), keepdims=True)
>>> res[0].shape
(1, 1, 1)  # axis=(0, 1, 2) is valid and returns array with `max` dims of 1s.
```

In the above example, `axis=(0, 1, 2)` is the longest valid axis tuple. If we assume `axis=None` to be equivalent to it, `max` rather than `min` behaves correctly.

> Can you find any two-"sample" NumPy functions we can use as precedent?

I too can't find any numpy function that takes more than one sample with different shapes. If you agree to the above argument, let's keep `max` otherwise I will change it to `min`. I am OK with both.

Reply:
Yup let's leave it as `max`. Good to make the decision explicitly, though!